mlflow/R/mlflow/R/model-registry.R

```diff
@@ -45,3 +121,194 @@ mlflow_delete_registered_model <- function(name, client = NULL) {
     data = list(name = forge::cast_string(name))
   )
 }
+
+#' List registered models
+#'
+#' Retrieves a list of registered models.
+#'
+#' @param max_results Maximum number of registered models to retrieve.
+#' @param page_token Pagination token to go to the next page based on a previous query.
+#' @template roxlate-client
+#' @export
+mlflow_list_registered_models <- function(max_results = 100, page_token = NULL, client = NULL) {
+  client <- resolve_client(client)
+
+  response <- mlflow_rest(
+    "registered-models",
+    "list",
+    client = client,
+    verb = "GET",
+    version = "2.0",
+    query = list(
+      max_results = max_results,
+      page_token = page_token
+    )
+  )
+
+  return(list(
+    registered_models = response$registered_model,
+    next_page_token = response$next_page_token
+  ))
+}
+
+#' Get latest model versions
+#'
+#' Retrieves a list of the latest model versions for a given model.
+#'
+#' @param name Name of the model.
+#' @param stages A list of desired stages. If the input list is NULL, return
+#'               latest versions for ALL_STAGES.
+#' @template roxlate-client
+#' @export
+mlflow_get_latest_versions <- function(name, stages = list(), client = NULL) {  # TODO: Fix multiple stage issue
+  client <- resolve_client(client)
+
+  response <- mlflow_rest(
+    "registered-models",
+    "get-latest-versions",
+    client = client,
+    verb = "GET",
+    version = "2.0",
+    query = list(
+      name = forge::cast_string(name),
+      stages = forge::cast_string_list(stages)
+    )
+  )
+
+  return(response$model_versions)
+}
+
+#' Create a model version
+#'
+#' @param name Register model under this name.
+#' @param source URI indicating the location of the model artifacts.
+#' @param run_id MLflow run ID for correlation, if `source` was generated by an experiment run in MLflow Tracking.
+#' @param tags Additional metadata.
+#' @param run_link MLflow run link - This is the exact link of the run that generated this model version.
+#' @param description Description for model version.
+#' @template roxlate-client
+#' @export
+mlflow_create_model_version <- function(name, source, run_id = NULL, tags = NULL, run_link = NULL, description = NULL, client = NULL) {
```

Comment:
I think for style reasons, we should keep 80 characters per line.

Reply:
Done.